File: python/chat-resource.py
from wetro import Wetrocloud
from decouple import config

WETRO_API_KEY = config("WETRO_API_KEY")

# Initialize the Wetrocloud client
client = Wetrocloud(api_key=WETRO_API_KEY)

# Create a chat history
chat_history = [
    {"role": "user", "content": "What is this collection about?"}, 
    {"role": "system", "content": "It stores research papers on AI technology."}
]

# Continue the conversation with context
chat_response = client.collection.chat(
    collection_id="louis",
    message="Can you explain the latest paper's methodology?",
    chat_history=chat_history
)

# Print the chat response
print(chat_response)

File: typescript/chat_resource.ts
import Wetrocloud from "wetro-sdk";
import dotenv from "dotenv";

// Load environment variables from .env file
dotenv.config();
const API_SECRET = process.env.WETRO_API_KEY;

// Initialize Wetrocloud with your API secret
const wetrocloud = new Wetrocloud({ apiSecret: API_SECRET || "" });

// Wrap everything in an async function
const run = async () => {
    try {
        const collection_id = "Kunle-badmus";
        const message = "What was the last thing you said";
        const chat_history: { role: "user" | "system"; content: string }[] = [
            { role: "user", content: "Hi there!" },
            { role: "system", content: "I am an AI assistant." },
            { role: "user", content: "Hello, how can you help me?" },
            { role: "system", content: "I can assist you with a variety of tasks, such as answering questions, providing information, helping with problem-solving, and offering recommendations. What do you need help with today?"}
        ];

        // Chat with the collection
        const response: any = await wetrocloud.chat({
            collection_id: collection_id,
            message: message,
            chat_history: chat_history,
        });

        console.log("Chat response:",response);
    } catch (error) {
        console.error("Error in chat:", error);
    }
};

// Run the async function
run();
